src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -310,6 +314,7 @@ public AmazonCorrettoCryptoProvider() {
         super("AmazonCorrettoCryptoProvider", PROVIDER_VERSION, "");
         this.relyOnCachedSelfTestResults = Utils.getBooleanProperty(PROPERTY_CACHE_SELF_TEST_RESULTS, true);
         this.shouldRegisterEcParams = Utils.getBooleanProperty(PROPERTY_REGISTER_EC_PARAMS, false);
+        this.shouldRegisterSecureRandom = Utils.getBooleanProperty(PROPERTY_REGISTER_SECURE_RANDOM, !isFips());
```

Comment:
An explanatory comment around the FIPS performance penalty would be helpful here. It's not immediately apparent why the defaulting is the way it is here.

Reply:
i hesitated to add one out of fear of it growing stale through time (AWS-LC's DRBG init latency may improve), but I agree that it's odd enough that it warrants a comment. will add.